kafka/src/main/java/org/apache/apex/malhar/kafka/KafkaConsumerWrapper.java

```diff
@@ -137,6 +189,9 @@ public void run()
 
 
         while (wrapper.isAlive) {
+          if (wrapper.waitForReplay) {
+            Thread.sleep(100);
+          }
```

Comment:
And also this will be an issue in old kafka operator that the thread could receive duplicated messages because it start consuming data immediately when the operator is activated. The offset it starts from will have overlap with offsets in the recovery window, correct?